Brain.pde

```diff
@@ -80,50 +83,106 @@ class Brain {
         }
       }
     }else{
-      neurons = null;
+      neurons = new float[BRAIN_WIDTH][BRAIN_HEIGHT];//null;
     }
   }
   public void useBrain(Creature owner){
-    ArrayList<Node> n = owner.n;
-    ArrayList<Muscle> m = owner.m;
+    final ArrayList<Node> n = owner.n;
+    final ArrayList<Muscle> m = owner.m;
     for(int i = 0; i < n.size(); i++){
       Node ni = n.get(i);
-      neurons[0][i] = dist(ni.x, ni.y, ni.z, foodX, foodY, foodZ);
+      neurons[0][i] = dist(ni.x, ni.y, ni.z, owner.foodX, owner.foodY, owner.foodZ);
     }
+    
     for(int i = 0; i < m.size(); i++){
       Muscle am = m.get(i);
       Node ni1 = n.get(am.c1);
       Node ni2 = n.get(am.c2);
       neurons[0][n.size()+i] = dist(ni1.x, ni1.y, ni1.z, ni2.x, ni2.y, ni2.z)/am.len;
     }
+    
+    neurons[0][n.size()+m.size()] = sin(heartbeatInterval*timesUsed);
+    neurons[0][n.size()+m.size()+1] = cos(2*heartbeatInterval*timesUsed);
+    timesUsed += 1;
+    
     for(int x = 1; x < BRAIN_WIDTH; x++){
-      for(int y = 0; y < BRAIN_HEIGHT-1; y++){
+      for(int y = 0; y < BRAIN_HEIGHT; y++){
         float total = 0;
         for(int input = 0; input < BRAIN_HEIGHT; input++){
           total += neurons[x-1][input]*axons[x-1][input][y].weight;
         }
-        if(x == BRAIN_WIDTH-1){
-          neurons[x][y] = total;
-        }else{
-          neurons[x][y] = sigmoid(total);
+        if(x == BRAIN_WIDTH - 1){
+          neurons[x][y] = sigmoidapprox(total);
+        } else {
+          neurons[x][y] = linear(total);
         }
       }
     }
+    for(int i = 0; i < n.size(); i++){
+      n.get(i).brainOutput = neurons[BRAIN_WIDTH-1][i];
+    }
     for(int i = 0; i < m.size(); i++){
       m.get(i).brainOutput = neurons[BRAIN_WIDTH-1][n.size()+i];
     }
   }
+  public float linear(float input){
+    if(input >= -5 || input <= 5){
```

Comment:
On this line in windows x64 I get the error: "processing.app.SketchException: Not expecting symbol 0xA0, which is NO-BREAK SPACE."

The offending character appears to be the whitespace between the or (`||`) and the second use of the `input` variable.

Deleting it and hitting the spacebar appears to resolve this issue.